File: tests_wraps/src/main/java/com/github/klee0kai/test/car/di/wrapped/custom/CarCustomWrappersComponent.java
package com.github.klee0kai.test.car.di.wrapped.custom;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.test.car.di.wrapped.custom.wrappers.*;
import com.github.klee0kai.test.car.model.Car;

@Component(
        wrapperProviders = {
                CarAsyncWrapper.class,
                CarWrapper.class,
        }
)
public interface CarCustomWrappersComponent {
    CarCustomWrapperModule module();

    Car car();

    CarRef<Car> carRef();

    CarLazy<Car> carLazy();

    CarProvide<Car> carProvide();

}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/car/di/wrapped/custom/CarCustomWrappersComponent.kt
package com.github.klee0kai.test.car.di.wrapped.custom

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.test.boxed.di.inject.CarBoxedWrapper
import com.github.klee0kai.test.car.di.wrapped.custom.wrappers.*
import com.github.klee0kai.test.car.model.Car

@Component(
    wrapperHelpers = [CarBoxedWrapper::class, CarRefWrapper::class],
)
interface CarCustomWrappersComponent {
    fun module(): CarCustomWrapperModule?
    fun car(): Car?
    fun carRef(): CarRef<Car?>?
    fun carLazy(): CarLazy<Car?>?
    fun carProvide(): CarProvide<Car?>?
}
